### PR TITLE
Add gear metadata to jobs and analyses

### LINKED ISSUE #1190: Denormalize some gear details on analyses
For #1189 Data Views - It would be great/essential to be able to filter analyses based on some gear details (e.g. id or name+version) in order to perform bulk file retrieval.

I propose adding those three fields at a minimum to analyses, with migration script.

### PR DESCRIPTION
**Pull request by ehlertjd**
Add the following fields to jobs and analyses on creation (and upgrade):
- `gear_name`
- `gear_version`
- `gear_category`

In addition, add the `gear_id` field to analysis.

I'm guessing it will be useful to be able to search some of these fields, but I haven't added them to the connector yet.

Closes #1190

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by hkethi002** [comment]
It might just be easier and more sane to include it, but won't the gear category be "analysis" for every analysis?

**Comment by ehlertjd** [comment]
@hkethi002  Today for sure. Don't know if that will stay the case forever, if someone else is confident that it will, it's pretty easily removed.

**Comment by kofalt** [comment]
It would be nice to isolate these keys under an object, which makes it a little clearer that it's a partial-join cache thing. Or, to put it another way, I will sigh sadly if we don't, haha. But no super-strong opinion

**Comment by ehlertjd** [comment]
@kofalt Done, I chose `gear_info` as the subdoc name

**Comment by ehlertjd** [comment]
@nagem I would agree with that - I was hesitant to call it `gear` on the subdoc because it's a super subset of the gear manifest.

**Comment by nagem** [comment]
>I'm guessing it will be useful to be able to search some of these fields, but I haven't added them to the connector yet.

They won't need to be added explicitly unless we want to change the format.

File: bin/database.py [hunk 1]
```diff
@@ -1726,6 +1726,41 @@ def upgrade_to_51():
...
     config.db.analyses.update_many({}, {"$unset": {"permissions": ""}})
+
+def upgrade_job_to_52(job, gears):
+    job_id = job.get('_id')
+    gear_id = str(job.get('gear_id'))
+    gear_doc = gears.get(gear_id)
+
+    if not gear_doc:
+        logging.warn('Unable to upgrade job {} to 52 - gear {} does not exist!'.format(job['_id'], gear_id))
+        return True
```
Comment (nagem) [inline bin/database.py]: Would we want to still save some form of an empty `gear_info` map? Or use the concept of the `dummy_gear` from `upgrade_to_22()`? Probably not, but I wanted to raise the option.

Comment (ehlertjd) [reply bin/database.py]: @nagem I guess I'd be inclined not to worry about it unless you can think of any value it would add.

**Pull request merged**